sdk/v2/core/projects_test.go

```diff
@@ -23,6 +23,81 @@ func TestProjectListMarshalJSON(t *testing.T) {
 	metaTesting.RequireAPIVersionAndType(t, ProjectList{}, "ProjectList")
 }
 
+func TestValidateProjectID(t *testing.T) {
+	testCases := []struct {
+		name       string
+		id         string
+		assertions func(err error)
```

Comment:
This is a common pattern that I like and use a lot, but we can do things a little more simply here. Instead of having each test case furnish a function that can make assertions, just have a bool and each test case can say whether the example is valid or not.